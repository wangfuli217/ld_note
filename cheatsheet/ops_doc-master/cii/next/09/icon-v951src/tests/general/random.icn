#SRC: JCON

#  test of random selection
#  (sets and tables are commented out because
#   v9 and jcon select different elements)

record r10(a,b,c,d,e,f,g,h,i,j)

procedure main()
   local r

   rtest()
   r := &random

   rtest()

   &random := 0
   rtest()

   &random := r
   rtest()

   &random := 11213
   rtest()
end

procedure rtest()
   local i
   static L, S, T, R
   initial {
      R := r10(1,2,3,4,5,6,7,8,9,10)
      L := []
      T := table()
      every i := 1 to 100 do {
         put(L, i)
	 T[i] := -i
      }
      S := set(L)
   }

   write()
   write(right(&random,10), "  i  ", ?9999)
   write(right(&random,10), "  i  ", ?9999)
   write(right(&random,10), "  r  ", integer(10000 * ?0))
   write(right(&random,10), "  r  ", integer(10000 * ?0))
   write(right(&random,10), "  s  ", ?"abcdefghijklmnopqrstuvwxyz")
   write(right(&random,10), "  s  ", ?"ABCDEFGHIJKLMNOPQRSTUVWXYZ")
   write(right(&random,10), "  c  ", ?&lcase)
   write(right(&random,10), "  c  ", ?&ucase)
   write(right(&random,10), "  L  ", ?L)
   write(right(&random,10), "  L  ", ?L)
   write(right(&random,10), "  R  ", ?R)
   write(right(&random,10), "  R  ", ?R)
#   write(right(&random,10), "  S  ", ?S)
#   write(right(&random,10), "  S  ", ?S)
#   write(right(&random,10), "  T  ", ?T)
#   write(right(&random,10), "  T  ", ?T)
   return
end
