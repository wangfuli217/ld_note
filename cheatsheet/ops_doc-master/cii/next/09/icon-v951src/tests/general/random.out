
         0  i  2114
 453816694  i  4124
 885666996  r  3157
 678165018  r  5104
1096161928  s  k
 905669982  s  H
 656467580  c  c
 170957890  c  T
1583830416  L  6
 108920774  L  72
1539632324  R  2
 295778538  R  4

 721762584  i  5331
1144737966  i  6208
1333202828  r  5762
1237514258  r  7075
1519504672  s  t
1583400982  s  G
 507287252  c  w
1824883130  c  I
 679975336  L  36
 764038654  L  90
1931954844  R  9
1814756834  R  1

         0  i  2114
 453816694  i  4124
 885666996  r  3157
 678165018  r  5104
1096161928  s  k
 905669982  s  H
 656467580  c  c
 170957890  c  T
1583830416  L  6
 108920774  L  72
1539632324  R  2
 295778538  R  4

 721762584  i  5331
1144737966  i  6208
1333202828  r  5762
1237514258  r  7075
1519504672  s  t
1583400982  s  G
 507287252  c  w
1824883130  c  I
 679975336  L  36
 764038654  L  90
1931954844  R  9
1814756834  R  1

     11213  i  1721
 369479103  i  7573
1626308041  r  3422
 734952971  r  9053
1944172069  s  z
2101642551  s  Q
1364863969  c  o
1221979715  c  E
 395152893  L  65
1380138031  L  20
 425567097  R  2
 407906811  R  2
